src/app/pages/home/rolodex/rolodex.page.ts

```diff
@@ -127,47 +125,34 @@ export class RolodexPage implements OnInit {
     }
   }
 
-  constructor(private authService: AuthService, private userService: UserService, private cardService: CardService,
-    private route: ActivatedRoute, private router: Router
+  constructor(
+    private authService: AuthService,
+    private userService: UserService,
+    private cardService: CardService,
+    private route: ActivatedRoute,
+    private router: Router
   ) {
-    console.log("Rolodex Page (constructor)");
+    console.log("Rolodex Page Start");
 
-    this.staticCards = this.cardService.listStaticCardsByHolder(this.authService.getCurrentUserId());
-    this.cards = this.cardService.listCardsByHolder(this.authService.getCurrentUserId());
   }
 
   ngOnInit() {
-    console.log("Rolodex Page (init)");
-    this.users = this.userService.listUsers();
-  }
-
-  deleteContact(index) {
-    //this.dataService.setContact(this.id, index, false);
-    this.refresh();
-  }
-
-  goToContactPage(index) {
-    this.index = index;
-    let navigationExtras: NavigationExtras = {
-      state: {
-        data: index,
-        parent: 'rolodex'
-      }
-    };
-    this.router.navigate(['contact'], navigationExtras);
-  }
-
-  refresh() {
-    //this.user = this.dataService.getUser(this.id);
-
-    // TODO: Page over this data when too large!
-    this.contactDataList = [];
-    for (let contactId in this.user.contacts) {
-      let contactData = {
-        id: contactId,
-        //name: this.dataService.getUser(contactId).name
-      };
-      this.contactDataList.push(contactData);
-    }
+    console.log("Rolodex Page Init");
+
+    let uid = this.authService.getCurrentUserId();
+    console.log('Current user ID: ', uid);
+
+    this.cardService.listStaticCardsByHolder(uid).pipe(
+      tap(data => {
+        console.log('Held static cards count: ', data.length);
```

Comment:
@brandlxee11 This logging wouldn't be possible if we simply used the Observable returned by `this.cardService.listStaticCardsByHolder(uid)`.